File: brightness.py
#!/usr/bin/env python

import sys, os

MAX_PATH = "/sys/class/drm/card0-LVDS-1/intel_backlight/max_brightness"
CUR_PATH = "/sys/class/drm/card0-LVDS-1/intel_backlight/brightness"

MAX = int(open(MAX_PATH, 'r').read().strip())
CUR = int(open(CUR_PATH, 'r').read().strip())
assert 0 <= CUR <= MAX <= 1e7

def usage():
    print "Usage: %s (up|down|[NUMBER])" % sys.argv[0]
    sys.exit(0)

def up():
    if CUR < 5:
        return CUR + 1
    if CUR < 25:
        return ((CUR + 7) // 5) * 5
    else:
        return int(CUR * 1.30)

def down():
    if CUR <= 5:
        return CUR - 1
    elif CUR <= 25:
        return ((CUR - 3) // 5) * 5
    else:
        return int(CUR / 1.30)

def main():
    if len(sys.argv) == 1:
        print CUR
        return

    if len(sys.argv) != 2:
        usage()

    cmd = sys.argv[1]
    if cmd == 'up':
        res = up()
    elif cmd == 'down':
        res = down()
    else:
        res = int(cmd)

    if res < 0:
        res = 0
    elif res > MAX:
        res = MAX

    print res
    os.execl("/usr/local/bin/_brightness", "_brightness", str(res))

if __name__ == "__main__":
    main()

File: brightness.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <string.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>

static const long int MAX_BRIGHTNESS = 3000;
static const char* PATH = "/sys/class/drm/card0-LVDS-1/intel_backlight/brightness";

int main (int argc, char **argv) {
  int fd;
  long int brightness;
  char *endptr;
  if(argc != 2 || argv[1] == NULL) {
    return EXIT_FAILURE;
  }

  if(strlen(argv[1]) == 0 || strlen(argv[1]) > 6) {
    return EXIT_FAILURE;
  }

  brightness = strtol(argv[1], &endptr, 10);

  if(strlen(endptr) != 0) {
    return EXIT_FAILURE;
  }

  if(brightness < 0 || brightness > MAX_BRIGHTNESS) {
    return EXIT_FAILURE;
  }

  setuid(0);
  setgid(0);
  if ((fd = open(PATH, O_WRONLY)) == -1) {
    return EXIT_FAILURE;
  }
  write(fd, argv[1], strlen(argv[1]));
  close(fd);
  return EXIT_SUCCESS;
}
